### PR TITLE
Add support Maccatalyst for MAUI

### PR DESCRIPTION
**Pull request by DmitriyKirakosyan**
* [x] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This PR adds support for Mac Catalyst.
Solution is based on https://github.com/microsoft/appcenter-sdk-dotnet/issues/1702/#issuecomment-1907191123.

## Related PRs or issues

[AB#96176](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/96176)
https://github.com/microsoft/appcenter-sdk-dotnet/issues/1702/

[Build link](https://dev.azure.com/msmobilecenter/SDK/_build/results?buildId=1520885&view=results)

### CODE REVIEW

File: SDK/Directory.Build.props [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <Copyright>Microsoft Corp. All rights reserved.</Copyright>
     <Company>Microsoft Corporation</Company>
     <!-- Pin C# version. C# 8 features might have issues on Mono runtime. -->
-    <LangVersion>7.0</LangVersion>
+    <LangVersion>8.0</LangVersion>
```
Comment (MikhailSuendukov) [inline SDK/Directory.Build.props:7]: The comment for this parameter says that version 8 may cause problems with Mono runtime. Has this changed?

Comment (DmitriyKirakosyan) [reply SDK/Directory.Build.props:7]: Now it doesn't even build with lang version 7 😅

**Comment by nor0x** [comment]
wtf 🤬

https://learn.microsoft.com/en-us/appcenter/retirement

**Comment by mhrastegari** [comment]
I knew they are waiting for it! 😅
But I wish they fix this for at least the last release! :)

**Review by lucen-ms — APPROVED** [review]

**Pull request merged**